addons/project/models/project.py

```diff
@@ -1807,6 +1822,11 @@ def write(self, vals):
         # rating on stage
         if 'stage_id' in vals and vals.get('stage_id'):
             self.filtered(lambda x: x.project_id.rating_active and x.project_id.rating_status == 'stage')._send_task_rating_mail(force_send=True)
+            if self.stage_id.sms_template_id and self.partner_id:
+                self._message_sms_with_template(
```

Comment:
This method needs a singleton only, so if the self contains more than one record, the method will raise an error.